expression/builtin_time_vec.go

```diff
@@ -541,11 +541,39 @@ func (b *builtinSubDateStringStringSig) vecEvalTime(input *chunk.Chunk, result *
 }
 
 func (b *builtinQuarterSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinQuarterSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		if err := handleInvalidTimeError(b.ctx, err); err != nil {
+			return err
+		}
```

Comment:
We need not call `handleInvalidTimeError` here, it's `args[0].VecEvalTime`'s responsibility to handle time error line by line.

Reply:
Thanks, I' ve fix it.